File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):

    count = 0
    visited = set()

    def dfs(node):
        visited.add(node)
        for j in range(n):
            if j not in visited and computers[node][j] == 1:
                dfs(j)

    for i in range(n):
        if i not in visited:
            dfs(i)
            count += 1
    
    return count

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    const visited = new Set();
    
    function bfs(start){
        const queue = [];
        queue.push(start);
        visited.add(start);
        
        while(queue.length > 0){
            const cur = queue.shift();
            
            for(let next = 0 ; next < n ; next++){
                if(visited.has(next) || computers[cur][next] === 0)continue;
                
                queue.push(next);
                visited.add(next);
            }
        }
        
        return 1;
    }
    
    for(let i = 0 ; i < n ; i++){
        if(visited.has(i)) continue;
        answer += bfs(i)
    }
    
    return answer;
}